### PR TITLE
[Issue-3461] Update UI popup update version

### PR DESCRIPTION
**Pull request by Quangdm-cdm**
PR for issue #3461

### CODE REVIEW

**Comment by saltict** [comment]
🚀 Deployed on https://pr-3670--sw-web-runner.netlify.app

File: packages/extension-koni-ui/src/Popup/Home/Earning/EarningEntry/EarningOptions/index.tsx [hunk 1]
```diff
@@ -360,6 +361,13 @@ function Component ({ className, hasEarningPositions, setEntryView }: Props) {
         subHeaderPaddingVertical={true}
         title={t<string>('Earning options')}
       >
+        <div className={'earning-banner-wrapper'}>
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Home/Earning/EarningEntry/EarningOptions/index.tsx:364]: move this div to !!banners.length && ( )

File: packages/extension-koni-ui/src/Popup/Home/Earning/EarningEntry/EarningPositions/index.tsx [hunk 2]
```diff
@@ -369,6 +371,13 @@ function Component ({ className, earningPositions, setEntryView, setLoading }: P
         subHeaderPaddingVertical={true}
         title={t<string>('Your earning positions')}
       >
+        <div className={'earning-banner-wrapper'}>
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Home/Earning/EarningEntry/EarningPositions/index.tsx:374]: move this div to !!banners.length && ( )

File: packages/extension-koni-ui/src/Popup/Home/Nfts/NftCollections.tsx [hunk 3]
```diff
@@ -163,20 +165,29 @@ function Component ({ className = '' }: Props): React.ReactElement<Props> {
...
-          minColumnWidth={'160px'}
-          renderItem={renderNftCollection}
-          renderOnScroll={true}
-          renderWhenEmpty={emptyNft}
-          searchFunction={searchCollection}
-          searchMinCharactersCount={2}
-          searchPlaceholder={t<string>('Search collection name')}
-        />
+        <>
+          <div className={'nft-banner-wrapper'}>
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Home/Nfts/NftCollections.tsx:169]: move this div to !!banners.length && ( )

File: packages/extension-koni-ui/src/Popup/Home/Tokens/index.tsx [hunk 4]
```diff
@@ -260,6 +261,13 @@ const Component = (): React.ReactElement => {
             />
           )
         }
+        <div className={'token-banner-wrapper'}>
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Home/Tokens/index.tsx:264]: move this div to !!banners.length && ( )

File: packages/extension-koni-ui/src/Popup/Settings/MissionPool/MissionDetailModal.tsx [hunk 5]
```diff
@@ -49,9 +54,22 @@ function Component ({ className = '', data }: Props): React.ReactElement<Props>
...
-    e.stopPropagation();
-    data?.url && openInNewTab(data.url)();
-  }, [data?.url]);
+    if (currentConfirmations && currentConfirmations.length) {
+      mktCampaignModalContext.openModal({
+        type: 'confirmation',
+        title: currentConfirmations[0].name,
+        message: currentConfirmations[0].content,
+        externalButtons: renderConfirmationButtons(mktCampaignModalContext.hideModal, () => {
+          e.stopPropagation();
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Settings/MissionPool/MissionDetailModal.tsx:63]: remove e.stopPropagation(); it is not necessary

File: packages/extension-koni-ui/src/Popup/Settings/MissionPool/index.tsx [hunk 6]
```diff
@@ -191,6 +202,14 @@ const Component: React.FC<Props> = ({ className }: Props) => {
       title={t<string>('Mission Pools')}
     >
       <div className={'__tool-area'}>
+        <div className={'mission-pool-banner-wrapper'}>
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Settings/MissionPool/index.tsx:205]: !!banners.length &&

File: packages/extension-koni-ui/src/Popup/Settings/MissionPool/index.tsx [hunk 7]
```diff
@@ -31,7 +33,8 @@ const Component: React.FC<Props> = ({ className }: Props) => {
   const { activeModal } = useContext(ModalContext);
   const [currentSelectItem, setCurrentSelectItem] = useState<MissionInfo | null>(null);
   const { missions } = useSelector((state: RootState) => state.missionPool);
-
+  const { banners, dismissBanner, onClickBanner } = useGetBannerByScreen('missionPools');
+  const { getCurrentConfirmation, renderConfirmationButtons } = useGetConfirmationByScreen('missionPools');
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/Popup/Settings/MissionPool/index.tsx:37]: move to MissionDetailModal

File: packages/extension-koni-ui/src/contexts/MktCampaignModalContext.tsx [hunk 8]
```diff
@@ -60,16 +67,16 @@ export const AppPopupModalContextProvider = ({ children }: AppPopupModalContextP
   }, [inactiveModal]);
 
   return (
-    <AppPopupModalContext.Provider value={{ openAppPopupModal, hideAppPopupModal }}>
+    <MktCampaignModalContext.Provider value={{ openModal, hideModal }}>
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/contexts/MktCampaignModalContext.tsx:70]: useMemo

File: packages/extension-koni-ui/src/hooks/campaign/useGetConfirmationByScreen.tsx [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+const useGetConfirmationByScreen = (screen: string) => {
+  const { appConfirmationMap, checkPositionParam, updateConfirmationHistoryMap } = useContext(AppOnlineContentContext);
+
+  const confirmations = useMemo(() => {
+    return appConfirmationMap[screen] || [];
+  }, [appConfirmationMap, screen]);
+
+  const getCurrentConfirmation = useCallback(
+    (compareVals: string[]) => {
+      return confirmations.filter((item) => {
```
Comment (lw-cdm) [inline packages/extension-koni-ui/src/hooks/campaign/useGetConfirmationByScreen.tsx:17]: find

**Review by lw-cdm — APPROVED** [review]

**Pull request merged**